ludwig/config_validation/validations.py

```diff
@@ -0,0 +1,184 @@
+"""Validation checks that are not easily covered by marshmallow schemas like parameter interdependencies.
+
+As these are built out, these auxiliary validations can be gradually removed.
+"""
+from threading import Lock
+from typing import Any, Dict, List
+
+from jsonschema import validate
+
+from ludwig.api_annotations import DeveloperAPI
+from ludwig.constants import (
+    COMBINED,
+    LOSS,
+    MODEL_ECD,
+    MODEL_TYPE,
+    NAME,
+    OUTPUT_FEATURES,
+    PREPROCESSING,
+    SPLIT,
+    TRAINER,
+    TYPE,
+)
+from ludwig.features.feature_registries import output_type_registry
+from ludwig.schema import get_schema, get_validator
+
+VALIDATION_LOCK = Lock()
+
+
+def check_feature_names_unique(config: Dict[str, Any]) -> None:
```

Comment:
@justinxzhao In our huddle you mentioned that you might not implement these first three checks because they're coming soon in other forms. This is true, but I would still like to see them validated here (for now) if its not a big hassle.

In particular, taking `check_feature_names_unique` as an example, the proposed fix you might have seen (https://github.com/ludwig-ai/ludwig/pull/2792) a couple days ago will only work for _frontend validation_, whereas the check your adding here would also be a value add for native Ludwig users.

The proposed solutions for `tied` and `dependent` features are what @connor-mccorm and I are trying to do via JSON schema but they will still be limited in scope, whereas these checks can be as extensive (and readable 😅 ) as necessary.

Reply:
Sounds good. I can follow up with PR(s) to implement most of the auxiliary checks.